backend/prisma/schema.prisma

```diff
@@ -8,20 +8,38 @@ datasource db {
 }
 
 model User {
-  id           String    @id @default(uuid())
-  email        String   @unique
-  firstName    String
-  lastName     String
-  password     String?
-  birthDate    DateTime?
-  googleID     String?   @unique
+  UserID      String             @id @default(uuid())
+  email       String             @unique
+  firstName   String
+  lastName    String
+  password    String?
+  birthDate   DateTime?
+  googleID    String?            @unique
   salt        String?
-  facebookID String?   @unique
-  venmoUrl String? 
-  cashappUrl String? 
-  paypalUrl String?
-  profile Profile?
+  facebookID  String?            @unique
+  venmoUrl    String?
+  cashappUrl  String?
+  paypalUrl   String?
+  profile     Profile?
+  contacts    Contact[]
+  User        UserRelationship[] @relation("self")
+  friendsWith UserRelationship[] @relation("friendsWith")
+  Pledge      Pledge[]
+}
+
+model UserRelationship {
+  UserRealtionshipID String             @id @default(uuid())
+  User               User               @relation("self", fields: [userID], references: [UserID])
+  userID             String
+  friendsWith        User               @relation("friendsWith", fields: [friendsWithID], references: [UserID])
+  friendsWithID      String
+  relationshipStatus RelationshipStatus @default(NOT_FRIEND)
+}
 
+enum RelationshipStatus {
+  FRIEND
+  PENDING_REQUEST
+  NOT_FRIEND
 }
```

Comment:
while it is legal syntax to leave enum like this, it sets the enum value to be numeric.

`RelationshipStatus.FRIEND` will have a value of `1`
which is why we generally try to do 

`FRIEND = 'friend'`

https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
same with all the other enums below

Reply:
Oh this is not enumtypescript. It's prisma way of defining enum for the ENUM constraint.
https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-enums

Reply:
ah in that case, let's still make sure these enum's don't drift from the typescript ones we'll have to use as well

https://github.com/prisma/prisma/discussions/9215

Reply:
It does make me wonder though what actually gets saved into the database? I feel like i'd be a number still right? I think it's fine to leave it this way (though I would recommend finding a way to maintain a single source of truth if possible). Do you know the answer?

Reply:
Yeah it does show up in the DB as `FRIEND`

![image](https://user-images.githubusercontent.com/106122976/224882300-3a43f24a-ee56-4ce4-ac06-9d97eecc0ccb.png)